cpp/libcudf_kafka/CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+    endif(GTEST_FOUND)
+endif(BUILD_TESTS)
+        
+message(STATUS "CUDF_KAFKA_TEST_LIST set to: ${CUDF_KAFKA_TEST_LIST}")
+
+###################################################################################################
+# - include paths ---------------------------------------------------------------------------------
+
+if(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES)
+	include_directories("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}")
+endif(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES)
+
+include_directories("${CMAKE_BINARY_DIR}/include"
+                    "${CMAKE_BINARY_DIR}/include/jit"
+                    "${CMAKE_SOURCE_DIR}/include"
+                    "${CMAKE_SOURCE_DIR}/src")
+
+if(CONDA_INCLUDE_DIRS)
+    include_directories("${CONDA_INCLUDE_DIRS}")
+endif(CONDA_INCLUDE_DIRS)
+
+###################################################################################################
+# - library paths ---------------------------------------------------------------------------------
+
+link_directories("${CMAKE_BINARY_DIR}/lib"
+                 "${CMAKE_BINARY_DIR}"
+                 "${GTEST_LIBRARY_DIR}"
+                 "${RMM_LIBRARY}")
+
+if(CONDA_LINK_DIRS)
+    link_directories("${CONDA_LINK_DIRS}")
+endif(CONDA_LINK_DIRS)
+
+###################################################################################################
+# - library target --------------------------------------------------------------------------------
+
+add_library(cudf_kafka SHARED
+    src/kafka_consumer.cpp
+)
+
+# Include paths
+include_directories("${CMAKE_SOURCE_DIR}/include"
+                    "${CMAKE_CURRENT_SOURCE_DIR}/include/cudf")
+
+###################################################################################################
+# cudf_kafka - librdkafka -------------------------------------------------------------------------
+
+find_path(RDKAFKA_INCLUDE "librdkafka" HINTS "$ENV{RDKAFKA_ROOT}/include")
+find_library(RDKAFKA++_LIBRARY "rdkafka++" HINTS "$ENV{RDKAFKA_ROOT}/lib" "$ENV{RDKAFKA_ROOT}/build")
+
+message(STATUS "RDKAFKA: RDKAFKA++_LIBRARY set to ${RDKAFKA++_LIBRARY}")
+message(STATUS "RDKAFKA: RDKAFKA_INCLUDE set to ${RDKAFKA_INCLUDE}")
+
+target_link_libraries(cudf_kafka ${RDKAFKA++_LIBRARY})
+include_directories("${RDKAFKA_INCLUDE}")
+
+###################################################################################################
+# - cudf_kafka Install ----------------------------------------------------------------------------
+target_link_libraries(cudf_kafka cudf)
+
+install(TARGETS cudf_kafka
+        DESTINATION lib
+        COMPONENT cudf_kafka)
+
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include
+        DESTINATION include
+        COMPONENT cudf_kafka)
+
+add_custom_target(build_tests_libcudf_kafka
+                    DEPENDS ${CUDF_KAFKA_TEST_LIST})
+
+add_custom_target(test_libcudf_kafka
+                    COMMAND ctest
+                    DEPENDS build_tests_libcudf_kafka)
```

Comment:
Instead of adding these custom targets and the COMPONENT labels, we should just use `make` / `make install` to build / install as needed.